packages/core/strapi/lib/services/entity-service/index.js

```diff
@@ -65,13 +65,15 @@ const createDefaultImplementation = ({ strapi, db, eventHub, entityValidator })
   },
 
   async findMany(uid, opts) {
-    const { kind } = strapi.getModel(uid);
-
     const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' });
 
     const query = transformParamsToQuery(uid, wrappedParams);
 
+    const { kind } = strapi.getModel(uid);
     if (kind === 'singleType') {
+      if (opts?.locale === 'all') {
+        return db.query(uid).findMany(query);
+      }
```

Comment:
It's something like that to do indeed! But it's not at the right place as i18n is a plugin and thus, no i18n logic should exist outside of the plugin.
You probably want to update this file instead: `packages/plugins/i18n/server/services/entity-service-decorator.js`.

I think I found something that works. Since I'm off next Monday and Tuesday, I suggest you try to find on your own how it works and how to do the fix. It's kind of tricky so if needed you can reveal my fix proposal that I can explain when I'm back:

<details>
  <summary>Pierre fix proposal</summary>
  
```js
const { transformParamsToQuery } = require('@strapi/utils').convertQueryParams;

...


  /**
   * Find an entry or several if fetching all locales
   * @param {string} uid - Model uid
   * @param {object} opts - Query options object (params, data, files, populate)
   */
  async findMany(uid, opts = {}) {
    const model = strapi.getModel(uid);

    const { isLocalizedContentType } = getService('content-types');

    if (!isLocalizedContentType(model)) {
      return service.findMany.call(this, uid, opts);
    }

    const { kind } = strapi.getModel(uid);

    const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' });

    const query = transformParamsToQuery(uid, wrappedParams);

    if (kind === 'singleType') {
      if (opts[LOCALE_QUERY_FILTER] === 'all') {
        return strapi.db.query(uid).findMany(query);
      }
      return strapi.db.query(uid).findOne(query);
    }

    return strapi.db.query(uid).findMany(query);
  },
```
  
</details>